### PR TITLE
docs: Automatically add REGAPIC documentation where appropriate

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

**Comment by jskeet** [comment]
Sample of rendered documentation for Vision:

image

**Review by amanda-tarafa — APPROVED** [review]
LGTM with a few nits

File: tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs [hunk 1]
```diff
@@ -266,6 +266,29 @@ private static string TransformDocTemplate(ApiMetadata api, string text)
 and we recommend using a single instance across your entire application unless you have a particular need
 to configure multiple client objects separately.";
 
+            if (api.Transport == "grpc+rest" || api.Transport == "rest+grpc")
+            {
+                clientConstruction +=
+$@"
```
[the remark this answers is not in the document]
Comment (jskeet) [reply tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs:272]: Nope, very much not. (See later.)

File: tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs [hunk 2]
```diff
@@ -266,6 +266,29 @@ private static string TransformDocTemplate(ApiMetadata api, string text)
...
+
+### Using the REST (HTTP/1.1) transport
+
+This library defaults to performing RPCs using [gRPC](https://grpc.io/) using the binary [Protocol Buffer](https://protobuf.dev) wire format.
+However, it also supports using HTTP/1.1 and JSON, for situations where gRPC doesn't work as desired. (This is typically due to an incompatible proxy
+or other network issue.) To create a client using HTTP/1.1, specify a `RestGrpcAdapter` reference for the `GrpcAdapter` property in the client builder.
```
[the remark this answers is not in the document]
Comment (jskeet) [reply tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs:278]: It's "APIs where `api.Transport` is `"grpc+rest"` or `"rest+grpc"`, which might sound familiar ;)

File: tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs [hunk 3]
```diff
@@ -266,6 +266,29 @@ private static string TransformDocTemplate(ApiMetadata api, string text)
...
 
+            if (api.Transport == "grpc+rest" || api.Transport == "rest+grpc")
+            {
+                clientConstruction +=
+$@"
+
+### Using the REST (HTTP/1.1) transport
+
+This library defaults to performing RPCs using [gRPC](https://grpc.io/) using the binary [Protocol Buffer](https://protobuf.dev) wire format.
+However, it also supports using HTTP/1.1 and JSON, for situations where gRPC doesn't work as desired. (This is typically due to an incompatible proxy
```
[the remark this answers is not in the document]
Comment (jskeet) [reply tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs:277]: How do you feel about "supports the use of" instead?

File: tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jskeet) [reply tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs:272]: (If it was the `$@` that you mean, we need it to be verbatim to use multiple lines, and we need it to be interpolated to use the sample client name.)

[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs:272]: Ah, yes, don't worry, didn't realize this was a string on a cs file. Duh

File: tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs:277]: Yep, or just "supports HTTP/1.1 and JSON". Definetel a nit though.

File: tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs:278]: :)

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Pull request merged**